dev-tools/mage/pkg.go

```diff
@@ -117,17 +117,19 @@ func Package() error {
 	return nil
 }
 
+// isPackageTypeSelected returns true if SelectedPackageTypes is empty or if
+// pkgType is present on SelectedPackageTypes. It returns false otherwise.
 func isPackageTypeSelected(pkgType PackageType) bool {
-	if SelectedPackageTypes != nil {
-		selected := false
-		for _, t := range SelectedPackageTypes {
-			if t == pkgType {
-				selected = true
-			}
+	if len(SelectedPackageTypes) == 0 {
+		return true
+	}
+
+	for _, t := range SelectedPackageTypes {
+		if t == pkgType {
+			return true
```

Comment:
Not exactly related to this change.
But the name of the function `isPackageTypeSelected` and the behavior of this method looks reversed to me.
It returns **true** when `SelectedPackageTypes` is **nil** or slice is **empty?**

Reply:
yes, it does. However from what I could understand, an empty/nil `SelectedPackageTypes` means, the environment variable `PACKAGES` isn't defined, therefore all package types are selected. Thus the function returning true when `SelectedPackageTypes` is either nil or empty.

Reply:
That's is counter intuitive, so I added the docs here and on the `SelectedPackageTypes` as well.

Reply:
Yes, agree, my brain did not compute correctly the `func` name, thanks for adding clarification.